main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification2.scala

```diff
@@ -16,26 +16,24 @@
 package ca.uwaterloo.flix.language.phase.unification
 
 import ca.uwaterloo.flix.api.Flix
-import ca.uwaterloo.flix.language.ast.{Rigidity, RigidityEnv, SourceLocation, Symbol, Type, TypeConstructor}
+import ca.uwaterloo.flix.language.ast.Ast.AssocTypeConstructor
+import ca.uwaterloo.flix.language.ast.{Kind, Rigidity, RigidityEnv, SourceLocation, Symbol, Type, TypeConstructor}
 import ca.uwaterloo.flix.language.phase.unification.FastBoolUnification.{ConflictException, Equation, Term, TooComplexException}
-import ca.uwaterloo.flix.language.phase.unification.UnificationError.TooComplex
-import ca.uwaterloo.flix.util.{InternalCompilerException, Result}
 import ca.uwaterloo.flix.util.collection.Bimap
-
-import scala.collection.mutable
+import ca.uwaterloo.flix.util.{InternalCompilerException, Result}
 
 object EffUnification2 {
 
   /**
     * Returns the most general unifier of the pairwise unification equations in `l`.
-   *
-   * @param l the list of unification equations.
-   * @param renv the rigidity environment.
-   * @param loc the source location of the entire equation system, e.g. the entire function body.
+    *
+    * @param l    the list of unification equations.
+    * @param renv the rigidity environment.
+    * @param loc  the source location of the entire equation system, e.g. the entire function body.
     */
   def unifyAll(l: List[(Type, Type, SourceLocation)], renv: RigidityEnv, loc: SourceLocation)(implicit flix: Flix): Result[Substitution, UnificationError] = {
     // Compute a bi-directional map from type variables to ints.
-    implicit val bimap: Bimap[Type.Var, Int] = mkBidirectionalVarMap(l)
+    implicit val bimap: Bimap[Atom, Int] = mkBidirectionalVarMap(l)
```

Comment:
We could do an experiment where we call the current Boolean solver here. 

Assuming it does not run out of resources that could tell us whether the two solvers agree?